game/board.py

```diff
@@ -324,3 +347,8 @@ def filter_move(self, dictionary):
             return CORRECT_MOVE
         else:
             return self.make_move(dictionary)
+
+    def item_quantity(self, item):
+        gold_quantity = sum([1 for row in self._board for cell in row if cell.gold > 0])
+        hole_quantity = sum([1 for row in self._board for cell in row if cell.has_hole])
```

Comment:
Could you combine this logic and iterate the list only once instead? Even though the complexity is O(^2) you're doing it twice -> O(^2) + O(^2)
If you see the iteration is the same `sum([1 for row in self._board for cell` the only thing that changes is the condition

Reply:
I could do it like that:

    def item_quantity(self, item):

        gold_quantity = 0
        hole_quantity = 0
        for row in self._board:
            for cell in row:
                gold_quantity += cell.gold > 0
                hole_quantity += cell.has_hole
        return hole_quantity if item == HOLE else gold_quantity